### ISSUE TITLE
custom dataset training and inference on new out of dataset images

### ISSUE THREAD

**Issue by papasanimohansrinivas** [zylo117/Yet-Another-EfficientDet-Pytorch#515]
![download (18)](https://user-images.githubusercontent.com/13809378/95487015-68614a80-09b1-11eb-8a79-a4e98f247eff.png)![download (15)](https://user-images.githubusercontent.com/13809378/95488551-508ac600-09b3-11eb-95f5-c0caa1ef5f52.png)

hi @zylo117  thanks for providing code for efficientdet first, i am training fire detection model with one class (fire) , using pretrained model with --head only argument. I am training it for **150 epochs** , **adamw optmizer** .I am doing inference of model and I am manually curating  results on trained and val dataset i.e by adjusting threshold for each image and verifying boxes.So for training and validation looks good but when i do it on **cctv footage of fire** , bounding boxes are all over the place like in above image.**Can u suggest something for improving results on unseen images**.Also my sample  training log 

**Val. Epoch**: 156/300. **Classification loss: 0.54603**. **Regression loss: 3.32129**. **Total loss: 3.86732**
Step: 3475. Epoch: 157/300. Iteration: 22/22. Cls loss: 0.08246. Reg loss: 0.23278. Total loss: 0.31524: 100% 22/22 [01:02<00:00,  2.84s/it]
Step: 3497. Epoch: 158/300. Iteration: 22/22. Cls loss: 0.04958. Reg loss: 0.14554. Total loss: 0.19512: 100% 22/22 [01:02<00:00,  2.83s/it]
Epoch    55: reducing learning rate of group 0 to 1.0000e-08.
Step: 3499. Epoch: 159/300. Iteration: 2/22. Cls loss: 0.07568. Reg loss: 0.14952. Total loss: 0.22519:   5% 1/22 [00:17<05:23, 15.41s/it]checkpoint...
Step: 3519. Epoch: 159/300. Iteration: 22/22. Cls loss: 0.05326. Reg loss: 0.11107. Total loss: 0.16434: 100% 22/22 [01:02<00:00,  2.85s/it]
Step: 3541. Epoch: 160/300. Iteration: 22/22. Cls loss: 0.08980. Reg loss: 0.27886. Total loss: 0.36866: 100% 22/22 [01:02<00:00,  2.83s/it]
Step: 3547. Epoch: 161/300. Iteration: 6/22. Cls loss: 0.08102. Reg loss: 0.14596. Total loss: 0.22698:  27% 6/22 [00:27<01:13,  4.62s/it]

**Comment by zylo117** [comment]
I think you should increase the number of both training and validation images to get a more accurate result. Because it seems to me that your model could be overfitting and the loss is very unstable.
